galasa-parent/dev.galasa.framework.api.cps/src/main/java/dev/galasa/framework/api/cps/internal/common/PropertyUtilities.java

```diff
@@ -0,0 +1,243 @@
+/*
+ * Copyright contributors to the Galasa project
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package dev.galasa.framework.api.cps.internal.common;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.validation.constraints.NotNull;
+
+import com.google.gson.Gson;
+
+import dev.galasa.framework.api.common.InternalServletException;
+import dev.galasa.framework.api.common.ServletError;
+import dev.galasa.framework.spi.ConfigurationPropertyStoreException;
+import dev.galasa.framework.spi.FrameworkException;
+import dev.galasa.framework.spi.IFramework;
+import dev.galasa.framework.spi.utils.GalasaGsonBuilder;
+
+import static dev.galasa.framework.api.common.ServletErrorMessage.*;
+
+public class PropertyUtilities {
+
+    private static final Set<String> updateActions = Set.of("apply","update");
+
+    static final Gson gson = GalasaGsonBuilder.build();
+    
+    IFramework framework;
+
+    private static final String REDACTED_PROPERTY_VALUE = "********";
+
+    private static final Set<String> hiddenNamespaces = new HashSet<>();
```

Comment:
Making this static means it's very hard to unit test things.
I hate statics in Java code, unless they are constants.

They should all be in objects, so unit tests can set them independently of each other.